ocaml/typing/uniqueness_analysis.ml

```diff
@@ -553,11 +1000,35 @@ module Usage_tree : sig
   (** Parallel composition lifted from [Usage.par]  *)
   val par : t -> t -> t
 
+  (** An empty tree containing only the root with empty usage *)
+  val empty : t
+
   (** A singleton tree containing only one leaf *)
-  val singleton : Usage.t -> Path.t -> t
+  val singleton : Usage.t -> Learned_tags.t -> Overwrites.t -> Path.t -> t
```

Comment:
As discussed, would be good to add a comment saying that pattern usages will have (optionally) learned tags, expression usages will have (optionally) overwrites, but never will a usage have both learned tags and overwrites.